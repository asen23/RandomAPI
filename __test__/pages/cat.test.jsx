import { render, screen } from "@testing-library/react"
import userEvent from "@testing-library/user-event"
import Cat from "pages/cat/index"

const dummyCat = [{ file: "file1.jpg" }, { file: "file2.jpg" }]

describe("Cat", () => {
    beforeEach(() => {
        fetch.resetMocks()
    })

    it("renders unchanged", async () => {
        fetch.mockResponseOnce(JSON.stringify(dummyCat[0]))

        const { asFragment } = render(<Cat />)
        expect(asFragment()).toMatchSnapshot()

        await screen.findByAltText("cat")

        expect(asFragment()).toMatchSnapshot()
    })

    it("gives new cat", async () => {
        fetch.mockResponses(...dummyCat.map((e) => JSON.stringify(e)))

        render(<Cat />)

        const firstImage = await screen.findByAltText("cat")
        expect(firstImage.src).toContain(dummyCat[0].file)

        const button = screen.getByRole("button")
        userEvent.click(button)

        const loading = await screen.findByText("Loading...")
        expect(loading).toBeDefined()

        const secondImage = await screen.findByAltText("cat")
        expect(secondImage.src).toContain(dummyCat[1].file)
    })
})
