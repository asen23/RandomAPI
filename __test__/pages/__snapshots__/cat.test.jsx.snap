// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cat renders unchanged 1`] = `
<DocumentFragment>
  <div
    class="flex justify-center items-center flex-col min-h-screen gap-8 bg-primary"
  >
    <h1
      class="text-4xl font-bold"
    >
      Random Cat
    </h1>
    <p>
      Loading...
    </p>
    <button
      class="border border-gray-400 px-2 py-1 rounded-lg hover:bg-white"
    >
      Get random cat
    </button>
  </div>
</DocumentFragment>
`;

exports[`Cat renders unchanged 2`] = `
<DocumentFragment>
  <div
    class="flex justify-center items-center flex-col min-h-screen gap-8 bg-primary"
  >
    <h1
      class="text-4xl font-bold"
    >
      Random Cat
    </h1>
    <img
      alt="cat"
      class="max-w-[80%] max-h-[60vh]"
      src="file1.jpg"
    />
    <button
      class="border border-gray-400 px-2 py-1 rounded-lg hover:bg-white"
    >
      Get random cat
    </button>
  </div>
</DocumentFragment>
`;
